### PR TITLE
adding templates for community involvement.

### PR DESCRIPTION
**Pull request by jhaynes**
### Summary
This adds an issue and pr template as well as contribution guidelines based off of the [aws/amazon-ecs-agent](github.com/aws/amazon-ecs-agent) repo.

This contribution is under the terms of the Apache 2.0 License: yes

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+### Summary
+<!-- What does this pull request do? -->
+
+### Implementation details
+<!-- How are the changes implemented? -->
+
+### Testing
+<!-- How was this tested? -->
+
+New tests cover the changes: <!-- yes|no -->
```
Comment (samuelkarp) [inline .github/PULL_REQUEST_TEMPLATE.md]: I don't think we have any tests here.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Summary
+<!-- What does this pull request do? -->
+
+### Implementation details
+<!-- How are the changes implemented? -->
+
+### Testing
+<!-- How was this tested? -->
+
```
Comment (samuelkarp) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can you add some checkboxes for different distributions?

Comment (jhaynes) [reply .github/PULL_REQUEST_TEMPLATE.md]: Do we support different distributions?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (jhaynes) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'm being aspirational.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (samuelkarp) [reply .github/PULL_REQUEST_TEMPLATE.md]: In README.md:

> Supported OS: Amazon Linux, Red Hat Enterprise Linux 7, Debian 8, Ubuntu 14.04

**Review by samuelkarp — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+### Testing
+<!-- How was this tested? -->
+- [ ] Works properly on Amazon Linux
+- [ ] Works properly on RHEL 7
+- [ ] Works properly on Debian 8
+- [ ] Works properly on Ubuntu 14.04
+
+New tests cover the changes: <!-- yes|no -->
+
+### Description for the changelog
```
Comment (jahkeup) [inline .github/PULL_REQUEST_TEMPLATE.md]: There isn't a `CHANGELOG` in this repo and this tool isn't versioned, I think this should be left out until that changes.

**Review by jahkeup — CHANGES_REQUESTED** [review]

**Review by jahkeup — APPROVED** [review]

**Pull request merged**